### PR TITLE
ISeeCube implentation

### PR DESCRIPTION
**Pull request by ChenLi2049**
This model requires [torchscale](https://github.com/microsoft/torchscale) 0.2.0, which can be installed by:

```Bash
pip install torchscale==0.2.0
```

I checked [Install — graphnet documentation](https://graphnet-team.github.io/graphnet/install.html), more information on IceTray would be much appreciated.

### CODE REVIEW

**Comment by RasmusOrsoe** [comment]
@ChenLi2049 Thanks for this! 

Quite a few of the tests fail for this PR; it appears to be mainly related to code formatting. Could you please install `pre-commit` (see [contribution guide](https://github.com/graphnet-team/graphnet/blob/main/CONTRIBUTING.md#code-quality)) and have it format the code?

**Comment by RasmusOrsoe** [comment]
@ChenLi2049 This still appears to be an issue (see https://github.com/graphnet-team/graphnet/actions/runs/8316406836/job/22883128660?pr=676).

**Comment by ChenLi2049** [comment]
Other than `Function __init__ has 9 arguments (exceeds 4 allowed). Consider refactoring.`, all the checks passed. I think it's good to go. :)

**Comment by RasmusOrsoe** [comment]
@ChenLi2049 thank you for dealing with the unit tests - glad to see you figured it out! Could you please update to the latest main and resolve the merge conflicts from #680?

**Comment by ChenLi2049** [comment]
If the modification of `FourierEncoder` is ok, it's ready to merge.

File: src/graphnet/models/components/embedding.py
Comment (RasmusOrsoe) [inline src/graphnet/models/components/embedding.py:64]: Let's set this `mlp_dim: Optional[int] = None`.

File: src/graphnet/models/components/embedding.py
Comment (RasmusOrsoe) [inline src/graphnet/models/components/embedding.py:74]: "mlp_dim (Optional): Size of hidden, latent space of MLP. If not given, `mlp_dim` defaults to `n_features`."

File: src/graphnet/models/components/embedding.py
Comment (RasmusOrsoe) [inline src/graphnet/models/components/embedding.py:95]:
```
if mlp_dim is None:
    mlp_dim = hidden_dim
```

File: src/graphnet/models/components/embedding.py
Comment (RasmusOrsoe) [inline src/graphnet/models/components/embedding.py:75]: "output_dim: Dimension of the output (I.e. number of columns)."

File: src/graphnet/models/gnn/icemix.py
Comment (RasmusOrsoe) [inline src/graphnet/models/gnn/icemix.py:37]: let's delete this new arg - we dont need it

File: src/graphnet/models/gnn/icemix.py
Comment (RasmusOrsoe) [inline src/graphnet/models/gnn/icemix.py:52]: Let's delete this too :-)

File: src/graphnet/models/gnn/icemix.py
Comment (RasmusOrsoe) [inline src/graphnet/models/gnn/icemix.py:71]: `mlp_dim = n_features`

**Review by RasmusOrsoe — COMMENTED** [review]
@ChenLi2049 thank you for this update.

We would want the Fourier-encoder to default to the settings that the 2nd place solution used to train their models with in the competition, such that we can utilize the pre-trained weights directly. I left a few comments along these lines. @ArturoLlorente can you confirm that this will be OK?

**Comment by ChenLi2049** [comment]
Hi @RasmusOrsoe, I am not familar with https://github.com/graphnet-team/graphnet/actions/runs/8707466377/job/23882513254?pr=676, can you help me with it?

**Comment by RasmusOrsoe** [comment]
@ChenLi2049 Hey - sorry for this. I have proposed a fix in #695 . Once it's merged into `main`, you should be able to resolve this error by updating your branch.

**Review by ArturoLlorente — DISMISSED** [review]
looks good! I think that it has no uncompatibilities with the IceMix implementation.

**Comment by RasmusOrsoe** [comment]
@ChenLi2049 issue is now fixed in `main`

**Comment by ChenLi2049** [comment]
Hi @ArturoLlorente , what kind of changes? I'm glad to help.

**Review by ArturoLlorente — COMMENTED** [review]
Hi @ChenLi2049, I approved the PR unintentionally. So I has to revert the approval with some comment.
However, I see that still some test failed. 
I think the easiest way to check it on your local computer will be to run:
`pre-commit run --all-files`

**Comment by ChenLi2049** [comment]
Hi @ArturoLlorente , [the failed check](https://github.com/graphnet-team/graphnet/actions/runs/8722392322/job/23928157381?pr=676) is failed when installing packages. It seems like the wrong package is downloaded?

**Comment by ChenLi2049** [comment]
OK, now other than "`__init__` has more than 4 arguments", all the checks passed. I think it's ready to merge. :)

**Review by ArturoLlorente — APPROVED** [review]
nothing else to add.  
nice work!

**Pull request merged**
